apps/tapipe.cxx

```diff
@@ -245,18 +245,18 @@ main(int argc, char* argv[])
   // Count number of TAs generated
   fmt::print("ta_buffer in bytes = {}\n", payload_size);
 
-  void* payload = malloc(payload_size);
+  char* payload = static_cast<char*>(malloc(payload_size));
 
 
   size_t offset(0);
   for ( const auto& ta : ta_buffer ) {
-    triggeralgs::write_overlay(ta, payload+offset);
+    triggeralgs::write_overlay(ta, static_cast<void*>(payload+offset));
     offset += triggeralgs::get_overlay_nbytes(ta);
   }
 
-  daqdataformats::Fragment ta_frag(payload, payload_size);
+  daqdataformats::Fragment ta_frag(static_cast<void*>(payload), payload_size);
 
-  free(payload);
+  free(static_cast<void*>(payload));
```

Comment:
Both here and in `EmulationUnit`, I would totally avoid using `malloc` and `free`. This is low-level C that should only be used if really cannot be avoided. In this case we could simply do:

```
  std::vector<char> payload(payload_size);

  size_t offset(0);
  for ( const auto& ta : ta_buffer ) {
    triggeralgs::write_overlay(ta, static_cast<void*>(payload.data()+offset));
    offset += triggeralgs::get_overlay_nbytes(ta);
  }
  daqdataformats::Fragment ta_frag(static_cast<void*>(payload.data()), payload_size);
```

And no `free` would be needed, since vector would go out of scope anyway (but could clear it if you must).


BUT this is not related to v4-to-v5 movement, so I'm fine if that's done at later date with a separate PR